vision.py

```diff
@@ -104,28 +162,48 @@ def _setup_cfg(config, opts, conf_thresh):
     return cfg
 
 
-def _get_video_from_tracks(tracks, images):
+def _get_video_from_tracks(tracks, images, output_file):
     ''' Save a video showing tracks to disk and return the path '''
-    
-    # fourcc = cv2.VideoWriter_fourcc(*'MP4V')
     output_size = images[0].shape
-    output_file = 'tracks.mp4'
-    # out = cv2.VideoWriter(output_file, fourcc, 25, output_size[:-1])
+    kelly_colors_rgb = [(255, 179, 0), (128, 62, 117), (255, 104, 0), (166, 189, 215),
+                    (193, 0, 32), (206, 162, 98), (129, 112, 102), (0, 125, 52),
+                    (246, 118, 142), (0, 83, 138), (255, 122, 92), (83, 55, 122),
+                    (255, 142, 0), (179, 40, 81), (244, 200, 0), (127, 24, 13),
+                    (147, 170, 0), (89, 51, 21), (241, 58, 19), (35, 44, 22)]
+
+    kelly_colors = [[x[2], x[1], x[0]] for x in kelly_colors_rgb]
+    
+    image_folder = "image_folder_{}".format(time.time())    
+    os.mkdir(image_folder)
     
-    for i in range(len(tracks)): # Number of tracks
+    image_files = []
+    for i in range(len(tracks[0])): # Number of tracks
         track_frame = np.zeros((output_size[0], output_size[1], 3), dtype=np.float32)
-        for j in range(len(tracks[0])): # Number of bounding boxes within a track
-            pt = tracks[i][j]
+
+        for j in range(len(tracks)): # Number of bounding boxes within a track
+            pt = tracks[j][i]
             if any(np.isnan(pt)):
                 continue
             x1, y1, x2, y2 = pt
             x, y, w, h = x1, y1, x2 - x1, y2 - y1 # Top left coordinates and width and height respectively
             cv2.rectangle(track_frame, (int(x), int(y)), (int(x + w), int(y + h)),
-                          (0, 255, 0), 10)
-        frame = np.where(True, images[i], track_frame)
+                          kelly_colors[j % len(kelly_colors)], 2)
+        
+        frame = np.where(track_frame != 0, track_frame, images[i])
         frame = Image.fromarray(frame.astype(np.uint8))
-        frame.save('image_folder/frame{}.jpg'.format(i)) 
-        # out.write(frame.astype(np.uint8))
-
-    # out.release()
+        image_file = '{}/frame{}.jpg'.format(image_folder, i)
+        image_files.append(image_file)
+        frame.save(image_file) 
+    
+    
+    clip = moviepy.video.io.ImageSequenceClip.ImageSequenceClip(image_files, fps=15)
```

Comment:
Why 15 fps?

Reply:
The sample video is exactly 30 frames long, and I didn't the tracking video to zoom by, so I thought that a 2 second video is appropriate